Source/WebCore/page/EventHandler.cpp

```diff
@@ -1934,14 +1934,18 @@ HandleUserInputEventResult EventHandler::handleMousePressEvent(const PlatformMou
         return true;
     }
 
+    RefPtr page = frame->page();
+    if (!page)
```

Comment:
maybe a debug assertion and/or logging might be useful to figure out the underlying issue eventually?

Reply:
A `WeakPtr<>`, by definition, could hold a `nullptr` value at any time, so it should always be checked for a `nullptr` value before being dereferenced.

If we do not expect it to be `nullptr`, we should use a different smart pointer type (maybe `CheckedPtr<>`), although this crash implies that a different smart pointer type would still crash.

Keeping the `nullptr` check and adding logging seems like a good plan if we want to change the `WeakPtr<>` to something else eventually.

Reply:
Yes, exactly. I agree with the ownership model here (page should be held as a WeakPtr), so a null check is in order but maybe no logging because we don't expect it to always stay non-null.